File: src/main/java/com/example/dpop/orchestrator/session/AuthenticationAttempt.java
package com.example.dpop.orchestrator.session;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

import java.time.Instant;
import java.util.UUID;

@Entity
@DiscriminatorValue("authentication")
public class AuthenticationAttempt extends OrchestratorAttempt {

    protected AuthenticationAttempt() {
    }

    public AuthenticationAttempt(UUID processSessionId, String nextContext, String nextStep, Instant expiresAt) {
        super(processSessionId, nextContext, nextStep, expiresAt);
    }
}

File: src/main/kotlin/com/example/dpop/orchestrator/session/AuthenticationAttempt.kt
package com.example.dpop.orchestrator.session

import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import java.time.Instant
import java.util.UUID

@Entity
@DiscriminatorValue("authentication")
class AuthenticationAttempt : OrchestratorAttempt {

    protected constructor() : super()

    constructor(
        processSessionId: UUID?,
        nextContext: String?,
        nextStep: String?,
        expiresAt: Instant?
    ) : super(processSessionId, nextContext, nextStep, expiresAt)
}
